tests/common/config_reload.py

```diff
@@ -61,10 +62,46 @@ def config_force_option_supported(duthost):
     return False
 
 
+def config_reload_with_minigraph_override(sonic_host, wait=120, start_bgp=True,
+                                          start_dynamic_buffer=True, safe_reload=False,
+                                          wait_before_force_reload=0, wait_for_bgp=False,
+                                          check_intf_up_ports=False, traffic_shift_away=False,
+                                          golden_config_path='/etc/sonic/golden_config_db.json',
+                                          is_dut=True, remote_src=False):
+    """This function is align with the purpose of minigraph deprecation. We cannot deprecate minigraph
+    used in sonic-mgmt directly. It provides a way to load extra config to DUT instead of modify the
+    attribute in minigraph.xml. It carries all param used in config_reload except below:
+
+    :param config_source: Pass 'minigrpah' as the override only imapct load_minigraph
+    :param override_config: Always True becuase of override
+    :param remote_src: Whether `src` is on the remote host or on the calling device.
+    """
+    BASE_DIR = os.path.dirname(os.path.realpath(__file__))
+    TEMPLATE_DIR = os.path.join(BASE_DIR, 'templates')
+    GOLDEN_CONFIG_TEMPLATE = 'golden_config_db.j2'
```

Comment:
>GOLDEN_CONFIG_TEMPLATE [](http://example.com/codeflow?start=4&length=22)

It will be more useful if client could change GOLDEN_CONFIG_TEMPLATE as a function parameter. In most cases, people just use the shared j2, but maybe some user want to use a separate one.